src/polyfem/State.hpp

```diff
@@ -492,389 +477,74 @@ namespace polyfem
-		double min_edge_length;
-		/// avg edge lenght
-		double average_edge_length;
-
-		/// errors, lp_err is in fact an L8 error
-		double l2_err, linf_err, lp_err, h1_err, h1_semi_err, grad_max_err;
-
-		/// non zeros and sytem matrix size
-		/// num dof is the total dof in the system
-		long long nn_zero, mat_size, num_dofs;
-
-		/// time to construct the basis
-		double building_basis_time;
-		/// time to load the mesh
-		double loading_mesh_time;
-		/// time to build the polygonal/polyhedral bases
-		double computing_poly_basis_time;
-		/// time to assembly
-		double assembling_stiffness_mat_time;
-		/// time to computing the rhs
-		double assigning_rhs_time;
-		/// time to solve
-		double solving_time;
-		/// time to compute error
-		double computing_errors_time;
-
-		/// statiscs on angle, compute only when using p_ref (false by default)
-		double max_angle;
-		/// statiscs on tri/tet quality, compute only when using p_ref (false by default)
-		double sigma_max, sigma_min, sigma_avg;
-
-		/// number of flipped elements, compute only when using count_flipped_els (false by default)
-		int n_flipped;
-
-		/// statiscs on the mesh (simplices)
-		int simplex_count;
-		/// statiscs on the mesh (regular quad/hex part of the mesh), see Polyspline paper for desciption
-		int regular_count;
-		/// statiscs on the mesh (regular quad/hex boundary part of the mesh), see Polyspline paper for desciption
-		int regular_boundary_count;
-		/// statiscs on the mesh (irregular quad/hex part of the mesh), see Polyspline paper for desciption
-		int simple_singular_count;
-		/// statiscs on the mesh (irregular quad/hex part of the mesh), see Polyspline paper for desciption
-		int multi_singular_count;
-		/// statiscs on the mesh (boundary quads/hexs), see Polyspline paper for desciption
-		int boundary_count;
-		/// statiscs on the mesh (irregular boundary quad/hex part of the mesh), see Polyspline paper for desciption
-		int non_regular_boundary_count;
-		/// statiscs on the mesh (irregular quad/hex part of the mesh), see Polyspline paper for desciption
-		int non_regular_count;
-		/// statiscs on the mesh (not quad/hex simplex), see Polyspline paper for desciption
-		int undefined_count;
-		/// statiscs on the mesh (irregular boundary quad/hex part of the mesh), see Polyspline paper for desciption
-		int multi_singular_boundary_count;
+	public:
+		/// Directory for output files
+		std::string output_dir;
 
 		/// flag to decide if exporting the time dependent solution to files
 		/// or save it in the solution_frames array
 		bool solve_export_to_file = true;
 		/// saves the frames in a vector instead of VTU
-		std::vector<SolutionFrame> solution_frames;
-
-		/// extracts the boundary mesh for visualization, called in build_basis
-		void extract_vis_boundary_mesh();
+		std::vector<output::SolutionFrame> solution_frames;
+		/// visualization stuff
+		output::OutGeometryData out_geom;
+		/// runtime statistics
+		output::OutRuntimeData runtime;
```

Comment:
`timings` or `running_times`? `runtime` sounds like it is runtime options or something when it really is just the running times.